packages/babel-types/src/definitions/typescript.ts

```diff
@@ -507,19 +507,42 @@ defineType("TSTypeAssertion", {
   },
 });
 
-defineType("TSEnumDeclaration", {
-  // "Statement" alias prevents a semicolon from appearing after it in an export declaration.
-  aliases: ["Statement", "Declaration"],
-  visitor: ["id", "members"],
+defineType("TSEnumBody", {
+  visitor: ["members"],
   fields: {
-    declare: validateOptional(bool),
-    const: validateOptional(bool),
-    id: validateType("Identifier"),
     members: validateArrayOfType("TSEnumMember"),
-    initializer: validateOptionalType("Expression"),
   },
 });
 
+if (process.env.BABEL_8_BREAKING) {
+  defineType("TSEnumDeclaration", {
+    // "Statement" alias prevents a semicolon from appearing after it in an export declaration.
+    aliases: ["Statement", "Declaration"],
+    visitor: ["id", "body"],
+    fields: {
+      declare: validateOptional(bool),
+      const: validateOptional(bool),
+      id: validateType("Identifier"),
+      // @ts-ignore(Babel 7 vs Babel 8) Babel 8 AST
+      body: validateType("TSEnumBody"),
+    },
+  });
+} else {
+  defineType("TSEnumDeclaration", {
+    // "Statement" alias prevents a semicolon from appearing after it in an export declaration.
+    aliases: ["Statement", "Declaration"],
+    visitor: ["id", "members"],
+    fields: {
+      declare: validateOptional(bool),
+      const: validateOptional(bool),
+      id: validateType("Identifier"),
+      members: validateArrayOfType("TSEnumMember"),
+      initializer: validateOptionalType("Expression"),
```

Comment:
The `initializer` field is also removed in Babel 8. The TS enum declaration does not support initializer (`enum { ... } = initializer`) and the babel parser does not use this field either. It probably refers to the `initializer` property in the `TSEnumMember`.